src/nl/rubensten/texifyidea/run/LatexCommandLineState.kt

```diff
@@ -73,14 +74,58 @@ open class LatexCommandLineState(environment: ExecutionEnvironment, private val
             return handler
         }
 
-        // Open Sumatra after compilation & execute inverse search.
-        if (runConfig.sumatraPath != null || isSumatraAvailable) {
-            SumatraForwardSearch().execute(handler, runConfig, environment)
+        // Do not open the pdf viewer when this is not the last run config in the chain
+        if (!runConfig.isLastRunConfig) {
+            return handler
         }
 
-        if(isEvinceAvailable()) {
+        // First check if the user specified a custom viewer, if not then try other supported viewers
+        if (!runConfig.viewerCommand.isNullOrEmpty()) {
+
+            // Split user command on spaces, then replace {pdf} if needed?
+            val commandString = runConfig.viewerCommand!!
+
+            // Split on spaces
+            val commandList = commandString.split(" ")
+
+            // Replace placeholder
+            var containsPlaceholder = false
```

Comment:
Use `val` instead of `var`. You're using a `map` with side effects. It's reads weirdly. You know what we mean ;)

Reply:
True, I have replaced it by a normal for loop now. But how am I supposed to modify a val? That's why it's a var.

Reply:
`val containsPlaceholder = commandList.contains("{pdf}")`? I don't think you need to modify it afterwards

Reply:
Ah I see, fixed that now, thanks!